### PR TITLE
Modding: allow mods to supply custom fonts

### PR DESCRIPTION
**Pull request by vegeta1k95**
This PR allows mods to supply custom fonts: `.ttf` files placed inside of `$modFolder/fonts/`.
Fonts will be visible and selectable in the font-list in `Advanced` options (and marked with mod name):

image

### CODE REVIEW

File: android/src/com/unciv/app/FontAndroid.kt [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                                Typeface.create(fontName, Typeface.NORMAL)
+                            }
+                        }
+                        else
+                        {
+                            Typeface.create(fontName, Typeface.NORMAL)
+                        }
+            }
+
+        }
```
Comment (yairm210) [inline android/src/com/unciv/app/FontAndroid.kt:77]: Far too much nesting!
- Reverse initial 'if' to 'if == then return'
- return at the end of 'if' of mod font
- reverse 'if' that includes the build-version

Alternatively, separate the system font into its own function, like `createSystemFont(fontName:String)`

**Review by yairm210 — APPROVED** [review]
Looks good it general!

File: android/src/com/unciv/app/FontAndroid.kt (hunk 1 above)
Comment (vegeta1k95) [reply android/src/com/unciv/app/FontAndroid.kt]: Changed

**Comment by yairm210** [comment]
I'm adding this in since I'm releasing a version, but I do want you to add documentation to the docs - currently this is a 'hidden feature' that modders won't know how to activate

**Pull request merged**